modular_skyrat/code/modules/research/xenoarch/strange_rock.dm

```diff
@@ -0,0 +1,262 @@
+			return
+		tryagain = FALSE
+	if(istype(W,/obj/item/xenoarch/clean/hammer/cm4))
+		if(tryagain)
+			to_chat(user,"You are already mining this.")
+			return
+		tryagain = TRUE
+		var/obj/item/xenoarch/clean/hammer/cm4/HM = W
+		playsound(loc, HM.usesound, 50, 1, -1)
+		if(!do_after(user,HM.cleandepth * 5,target = src))
+			to_chat(user,"You must stand still to clean.")
+			tryagain = FALSE
+			return
+		dugdepth += HM.cleandepth
+		if(dugdepth > itemactualdepth)
+			to_chat(user,"The strange rock crumbles, destroying anything that could have been recovered.")
+			qdel(src)
+			return
+		tryagain = FALSE
+	if(istype(W,/obj/item/xenoarch/clean/hammer/cm5))
+		if(tryagain)
+			to_chat(user,"You are already mining this.")
+			return
+		tryagain = TRUE
+		var/obj/item/xenoarch/clean/hammer/cm5/HM = W
+		playsound(loc, HM.usesound, 50, 1, -1)
+		if(!do_after(user,HM.cleandepth * 5,target = src))
+			to_chat(user,"You must stand still to clean.")
+			tryagain = FALSE
+			return
+		dugdepth += HM.cleandepth
+		if(dugdepth > itemactualdepth)
+			to_chat(user,"The strange rock crumbles, destroying anything that could have been recovered.")
+			qdel(src)
+			return
+		tryagain = FALSE
+	if(istype(W,/obj/item/xenoarch/clean/hammer/cm6))
+		if(tryagain)
+			to_chat(user,"You are already mining this.")
+			return
+		tryagain = TRUE
+		var/obj/item/xenoarch/clean/hammer/cm6/HM = W
+		playsound(loc, HM.usesound, 50, 1, -1)
+		if(!do_after(user,HM.cleandepth * 5,target = src))
+			to_chat(user,"You must stand still to clean.")
+			tryagain = FALSE
+			return
+		dugdepth += HM.cleandepth
+		if(dugdepth > itemactualdepth)
+			to_chat(user,"The strange rock crumbles, destroying anything that could have been recovered.")
+			qdel(src)
+			return
+		tryagain = FALSE
+	if(istype(W,/obj/item/xenoarch/clean/hammer/cm15))
+		if(tryagain)
+			to_chat(user,"You are already mining this.")
+			return
+		tryagain = TRUE
+		var/obj/item/xenoarch/clean/hammer/cm15/HM = W
+		playsound(loc, HM.usesound, 50, 1, -1)
+		if(!do_after(user,HM.cleandepth * 4,target = src))
+			to_chat(user,"You must stand still to clean.")
+			tryagain = FALSE
+			return
+		dugdepth += HM.cleandepth
+		if(dugdepth > itemactualdepth)
+			to_chat(user,"The strange rock crumbles, destroying anything that could have been recovered.")
+			qdel(src)
+			return
+		tryagain = FALSE
+	if(istype(W,/obj/item/xenoarch/clean/hammer/advanced))
+		if(tryagain)
+			to_chat(user,"You are already mining this.")
+			return
+		tryagain = TRUE
+		var/obj/item/xenoarch/clean/hammer/advanced/HM = W
+		playsound(loc, HM.usesound, 50, 1, -1)
+		if(!do_after(user,HM.cleandepth / 2,target = src))
+			to_chat(user,"You must stand still to clean.")
+			tryagain = FALSE
+			return
+		dugdepth += HM.cleandepth
+		if(dugdepth > itemactualdepth)
+			to_chat(user,"The strange rock crumbles, destroying anything that could have been recovered.")
+			qdel(src)
+			return
+		tryagain = FALSE
+	if(istype(W,/obj/item/xenoarch/clean/brush))
```

Comment:
Same for the brushes. Only difference is cleaning speed(?), could make that a variable of item/xenoarch/clean and save dozens of lines of duplicate code

Reply:
Addressed in a previous comment: Only doing hammers, not brushes atm.